### PR TITLE
Support for floating point predictor (type 3) with TIFF LZW Compression

### PR DESCRIPTION
**Pull request by Kontinuation**
This PR Resolves https://github.com/geosolutions-it/imageio-ext/issues/246

We also refactored the TIFFLZWDecompressor to use the `PredictorDecompressor` introduced by the [ZSTD support patch](https://github.com/geosolutions-it/imageio-ext/pull/245), now we don't have to maintain separate copies of predictors in decompressors.

I've also found that zstd_p3.tif was not correct, it was using predictor type 2, I've regenerated the TIFF to use predictor type 3.

### CODE REVIEW

**Comment by Kontinuation** [comment]
@aaime Please help review this PR when you have time :)

File: plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFLZWDecompressor.java [hunk 1]
```diff
@@ -130,29 +126,6 @@ public void decodeRaw(byte[] b,
                           int dstOffset,
                           int bitsPerPixel,
                           int scanlineStride) throws IOException {
-
-        // Check bitsPerSample.
-        if (predictor ==
```
Comment (aaime) [inline plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFLZWDecompressor.java:135]: This bit in in PredictorDecompressor.validate, called later

File: plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFLZWDecompressor.java [hunk 2]
```diff
@@ -233,84 +211,7 @@ public int decode(byte[] sdata, int srcOffset,
 	    }
 	}
 
-        if (predictor == BaselineTIFFTagSet.PREDICTOR_HORIZONTAL_DIFFERENCING) {
```
Comment (aaime) [inline plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFLZWDecompressor.java:236]: And this indeed is covered inside the PredictorDecompressor.decompress method, which also adds support for the floating point predictor.

**Review by aaime — APPROVED** [review]
Looks good to me.

**Pull request merged**